p2p/transport/webtransport/transport.go

```diff
@@ -268,3 +275,51 @@ func (t *transport) Close() error {
 	}
 	return nil
 }
+
+// extractSNI returns what the SNI should be for the given maddr. If there is an
+// SNI component in the multiaddr, then it will be returned and
+// foundSniComponent will be true. If there's no SNI component, but there is a
+// DNS-like compoent, then that will be returned for the sni and
+// foundSniComponent will be false (since we didn't find an actual sni component).
+func extractSNI(maddr ma.Multiaddr) (sni string, foundSniComponent bool) {
+	ma.ForEach(maddr, func(c ma.Component) bool {
+		switch c.Protocol().Code {
+		case ma.P_SNI:
+			sni = c.Value()
+			foundSniComponent = true
+			return false
+		case ma.P_DNS, ma.P_DNS4, ma.P_DNS6, ma.P_DNSADDR:
+			sni = c.Value()
+			// Keep going in case we find an `sni` component
+			return true
+		}
+		return true
+	})
+	return sni, foundSniComponent
+}
+
+// Resolve implements transport.Resolver
+func (t *transport) Resolve(ctx context.Context, maddr ma.Multiaddr) ([]ma.Multiaddr, error) {
+	sni, foundSniComponent := extractSNI(maddr)
```

Comment:
```suggestion
	sni, ok := extractSNI(maddr)
```

Reply:
Resolving this as won't fix since I'm not calling this `ok` for a reason.